### ISSUE TITLE
Make it possible to compose order transforms and query parameter ordering

### ISSUE THREAD

**Issue by dmos62** [centerofci/mathesar#2304]
## Problem

Brent [writes](https://github.com/centerofci/mathesar/pull/2253#pullrequestreview-1251718380):

> If there are multiple ordering steps (either Order transforms as part of the query, or order_by given as a query string parameter), the current setup will clobber any previous ordering with the next ordering step. The reason is that we create a deterministic ordering at each step, and we do this by adding the primary key (for example) as an ordering column. So, if step one orders by email domain, and step two orders by last name of a user, then there will be no trace of ordering by email domain in the final order.

In other words, it might be desirable for ordering to be implicitly composable, so that if you have ordering A and a subsequent ordering B, B should become the primary ordering, while A should implicitly become the secondary ordering.

## Proposed solution

I see three solutions:

- If there are multiple ordering being applied, one of them needs to be total, because we want our orderings to always be deterministic; the total ordering should always and only be the first ordering, subsequent orderings should not force totality;
- Fork the `Order` transform and the `order_by` query parameter into `TotalOrder`, `PartialOrder` and `total_order_by` and `partial_order_by`, thus letting the API client decide when an ordering directive should force totality.

**Comment by mathemancer** [comment]
I think we should bump this up in priority. As an example, it's not currently possible to make a schedule for a room in our Mathesar Con demo data set, since you'd need to order by the `day, time slot` pair. This is a bit blocking for example explorations, and I don't really think releasing without being able to have multiple orderings in the data explorer would be good.

**Comment by dmos62** [comment]
I'll add that we can make ordering transformations composable (to a considerable extent at least), but the UX will be a bit weird, because the user will be forced to think about ordering composability, defining sorts in the order of least primary to most primary (which is the reverse of what's idiomatic, I think). Ideally, we'd have sorting UI that allowed you to explicitly define the primary sort, and optionally the secondary, the tertiary, etc.

**Comment by kgodey** [comment]
I don't think this blocks release, but I put this in Backlog as High priority. 

> Ideally, we'd have sorting UI that allowed you to explicitly define the primary sort, and optionally the secondary, the tertiary, etc.

We're not aiming for ideal behavior right now, just good enough.
